File: etamen-backend/app/Modules/CarePlans/Infrastructure/Models/CarePlanDay.php
<?php

namespace App\Modules\CarePlans\Infrastructure\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class CarePlanDay extends Model
{
    protected $fillable = [
        'care_plan_id',
        'day_number',
        'day_date',
        'title',
        'instructions',
        'is_active',
    ];

    protected function casts(): array
    {
        return [
            'day_number' => 'integer',
            'day_date' => 'date',
            'is_active' => 'boolean',
        ];
    }

    public function plan(): BelongsTo
    {
        return $this->belongsTo(CarePlan::class, 'care_plan_id');
    }

    public function meals(): HasMany
    {
        return $this->hasMany(CarePlanMeal::class)->orderBy('sort_order');
    }
}

File: etamen-flutter/lib/features/care_plans/domain/entities/care_plan_day.dart
import 'package:etamen_app/features/care_plans/domain/entities/care_plan_meal.dart';

class CarePlanDay {
  const CarePlanDay({
    required this.id,
    required this.carePlanId,
    this.dayNumber,
    this.dayDate,
    this.title,
    this.instructions,
    this.isActive = true,
    this.meals = const [],
  });

  final int id;
  final int carePlanId;
  final int? dayNumber;
  final String? dayDate;
  final String? title;
  final String? instructions;
  final bool isActive;
  final List<CarePlanMeal> meals;
}
